vite-project/src/components/Form.tsx

```diff
@@ -0,0 +1,135 @@
+import "./Form.css";
+import React, { useState, useEffect, useRef } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrashAlt } from "@fortawesome/free-solid-svg-icons";
+
+interface Props {
+  onSubmit: (data: FormData) => void;
+}
+
+enum Genre {
+  Pop = "Pop",
+  Rock = "Rock",
+  HipHopRap = "Hip-Hop/Rap",
+  KPop = "K-Pop (Korean Pop)",
+  Funk = "Funk",
+  Metal = "Metal",
+  Jazz = "Jazz",
+}
+
+export { Genre };
+
+export type FormData = {
+  name: string;
+  genre: Genre;
+  artist: string;
+  date: string;
+};
+
+export default function Form({ onSubmit }: Props) {
+  const [formData, setFormData] = useState<FormData>({
+    name: "",
+    genre: Genre.Pop,
+    artist: "",
+    date: "",
+  });
+
+  const today = new Date().toISOString().split("T")[0];
+  const dateInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (dateInputRef.current) {
+      dateInputRef.current.max = today;
+    }
+  }, [today]);
```

Comment:
You used useEffect hook, very nice👍

But in this case I think you can do that much simplier
just create `today` constant which contains formatted string of todays date, and pass it to `max` prop